### PR TITLE
Correct behavior when erasing search input

### LINKED ISSUE #323: Unable to delete search input text with backspace after having executed full text search.
The input text reappears when trying to delete the leftmost (last remaining) character.

### LINKED ISSUE #379: Street search query stays in the input box
After closing the search / street

### PR DESCRIPTION
**Pull request by vaaralav**
Fixes #323

### CODE REVIEW

**Comment by tituomin** [comment]
Actually fixes #323, not #379.

File: src/views/navigation.coffee [hunk 1]
```diff
@@ -241,8 +241,8 @@ define [
 
         open: (event) ->
             @_open $(event.currentTarget).data('type')
-
-        toggleOnKeypress: (event) ->
+        # Didn't find out any use case for this function
```
Comment (tituomin) [inline src/views/navigation.coffee]: The function is actually needed. See line 221. It enables the user to select the Browse -functionality by pressing enter when the focus is on the Browse-header. This is important for accessibility, we have tried to enable focus+keyboard navigability as much as possible.

File: src/views/search-input.coffee [hunk 2]
```diff
@@ -53,9 +49,10 @@ define [
 
         search: (e) ->
             e.stopPropagation()
-            unless @isEmpty()
-                @$searchEl.typeahead 'close'
-                @executeQuery()
+            alert();
```
Comment (tituomin) [inline src/views/search-input.coffee]: Remove this.

File: src/views/navigation.coffee (hunk 1 above)
Comment (vaaralav) [inline src/views/navigation.coffee]: This makes it impossible to continue with different search phrase. Eg. one searches first for "päiväkotti" and then wants to correct the typo in the input field and search again for "päiväkoti". When they hit enter the input field for search gets completely erased and an empty search is made.

**Pull request merged**